src/datasets.c

```diff
@@ -542,10 +549,18 @@ static int DatasetLoadString(Dataset *set)
 
             // coverity[alloc_strlen : FALSE]
             uint8_t decoded[strlen(line)];
-            uint32_t consumed = 0, num_decoded = 0;
-            Base64Ecode code = DecodeBase64(decoded, (uint32_t)strlen(line), (const uint8_t *)line,
-                    (uint32_t)strlen(line), &consumed, &num_decoded, Base64ModeStrict);
-            if (code == BASE64_ECODE_ERR) {
+            uint32_t num_decoded = 0;
+            Base64Decoded *b64d =
+                    Base64Decode((const uint8_t *)line, strlen(line), 0, Base64ModeStrict);
+            if (b64d == NULL) {
+                FatalErrorOnInit("bad base64 encoding %s/%s", set->name, set->load);
+                continue;
+            }
+            DEBUG_VALIDATE_BUG_ON(b64d->decoded_len >= strlen(line));
+            memcpy(decoded, b64d->decoded, b64d->decoded_len);
```

Comment:
what I don't like here is that we have a stack allocated buffer "decoded", which we don't initially use, but instead we decode into a heap allocation, and copy that into the stack allocation.

Would rather see the stack allocated buffer to be passed to the `Base64Decode` function again, where it can be filled directly.

Reply:
ok. will change it